### DISCUSSION TITLE
feature: image uploading and media library

### DISCUSSION THREAD

**Discussion by airtonix** [Thinkmill/keystatic#380] [Ideas]
One of the really nice things that DecapCMS does is a media library, this is facilitated through two main aspects

- an image widget: https://decapcms.org/docs/widgets/#image
- collections or widgets can describe where images are stored: https://decapcms.org/docs/beta-features/#folder-collections-media-and-public-folder

essential:

- [x] handle upload one image

nice to have:

- [ ] handle upload multiple images 
- [ ] optimise images before uploading with [fengyuanchen/compressorjs](https://github.com/fengyuanchen/compressorjs)
- [ ] generate and store blurhash on upload as base64 string

--- 

[stormwarning/netlify-cms-widget-image-extra](https://github.com/stormwarning/netlify-cms-widget-image-extra)
[not-dalia/netlify-cms-widget-image-dimensions](https://github.com/not-dalia/netlify-cms-widget-image-dimensions)
[mvolfik/netlify-cms-widget-external-image](https://github.com/mvolfik/netlify-cms-widget-external-image)
Image Compression in browser:

[BosNaufal/react-image-compressor](https://github.com/BosNaufal/react-image-compressor)
[medium.com/front-end-weekly/image-compression-in-reactjs-a07ec0066b24](https://medium.com/front-end-weekly/image-compression-in-reactjs-a07ec0066b24)
[fengyuanchen/compressorjs](https://github.com/fengyuanchen/compressorjs)

**Comment by ikx94** [comment]
This would be great to have.
